### ISSUE TITLE
Scrum Update 4

### ISSUE THREAD

**Issue by DSep** [CourseFlo/Flo#27]
- This sprint I worked on creating the results section of the Browse page (frontend) connected to the redux store of search results. I also worked with Brandon to design a Course schema for our DB, then scripting an API call and data transform from a 3rd party api's course data for our own database. We also created the backend's course search API to make a complex query from the DB and then connected to it on the Browse page and components. I helped Guanting transition the VisualCourse page to be programmatically generated and connected to some basic redux. At the end Brandon and I also built a course fetch API for VisualCourse to retrieve a target course and its related courses and fixed the frontend to show them.
- Challenging parts of this sprint was the motivation to fix/improve the frontend and UI styles as a whole. We hit a wall in terms of motivation and are re-evaluating what our plans our and our stakes in this project. 
- We are still undergoing a project re-evaluation and discussion of our milestones so plans our not solid today. Generally, I'm hoping to do some of:
  - help with setting up our cloud deployment, 
  - possibly setup some CI/CD for that, 
  - support creating further endpoints that we might need in our app, 
  - look into adapting the UBC courses scraper to do our scraping of course data
  - improve the visualizer, possibly integrating a 3rd party visualization library as discussed in our milestones.

csid: g3u1b

**Comment by yqj2k** [comment]
- I worked on the user page. I connected the forms on the user page to persist data on the frontend, send it to the database, all asynchronously. This involved creating user endpoints, schema's/tables for mongoose, and asynchronous calls from the frontend to get and update user information.
- Debugging the front-end to connect to the database took a lot more time than necessary. I can reduce my time I am unblocked by giving more updates to my team members.
- I will refactor the profile page front-end components (previously having actions/reducers all in one file), and help out with deploying our application.

csid: a4g2b

**Comment by beetai** [comment]
- worked on creating a new homepage and worked with Sepand to create a script that populated our database with courses. Additionally, we worked together to create a functional basic search of the courses, which is done in the backend and wired up to the frontend of the Browse page.
- the learning curve of axios and thunk is something that I definitely struggled with when I first starting using it. Additionally, sorting out the details involving certain features also proved to be challenging sometimes.
- I plan to work on looking into heroku and deployment. I also want to work towards connecting all the pages together in terms of the features we had in mind (i.e. navbar search should redirect to the Browse page)

csid: c8m1b

**Comment by baconandchips** [comment]
- Worked on developing the visualizer page with my teammates, and had Redux set up with them. I also looked a bit into Material-UI, but that wasn't reflected in the sprint's codebase. 
- I struggled a fair bit with wrapping my head around Thunk and Redux still; it will be my mission this sprint to reduce technical debt with a video tutorial series.
- I plan to look into OAuth, and if possible help others on other components as well.
csid: n0k0b
